paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/SynchronizationActionBase.java

```diff
@@ -181,8 +181,13 @@ protected FileStoreTable alterTableOptions(Identifier identifier, FileStoreTable
         Map<String, String> withoutBucket = new HashMap<>(tableConfig);
         withoutBucket.remove(CoreOptions.BUCKET.key());
 
+        Map<String, String> oldOptions = table.options();
         List<SchemaChange> optionChanges =
                 withoutBucket.entrySet().stream()
+                        .filter(
+                                entry ->
+                                        !Objects.equals(
+                                                entry.getValue(), oldOptions.get(entry.getKey())))
```

Comment:
I think this is not enough because SchemaManager#checkAlterTableOption just forbids to commit changes that the key is immutable.

Reply:
done.